### ISSUE TITLE
Build/Install on ARM Issue

### ISSUE THREAD

**Issue by schnitzlein** [google/mediapipe#4591]
### OS Platform and Distribution

Linux Raspbian GNU/Linux 11 (bullseye)

### Compiler version

gcc/g++ version = 10.2.1

### Programming Language and version

Python 3.9.2

### Installed using virtualenv? pip? Conda?(if python)

virtualenv

### MediaPipe version

v0.9.3.0, v0.8.x

### Bazel version

4.2.2

### Android AAR (if android)

None

### OpenCV version (if running on desktop)

3.4

### Describe the problem

```shell
Hi,

I tested the google mediapipe and opencv with success under Ubuntu 22.04.
It works great and as expected. 
The https://github.com/google/mediapipe/blob/master/docs/solutions/hands.md works as really good.

Switching to Raspberry Pi 4 + google a lot of dependencies and other solutions.
I found out that I have to compile mediapipe with opencv as a dependency.
Python3 package is my goal so I searched for existing solutions.

[+] https://github.com/superuser789/MediaPipe-on-RaspberryPi#pre-built-packages
[+] https://spyjetson.blogspot.com/2021/06/installing-mediapipecpu-mode-on.html
[+] https://github.com/koenvervloesem/bazel-on-arm

Mostly some things work, some versions match together.
But bazel is always failing it try to download some file from:


To fix the issue with the newer/older mediapipe versions where the flag is missing --experimental_allow_proto3_optional , I just added it in the setup.py and generated the protobuf files succuessfully.
Opencv compiled succuessfully with or without gestreamer.

Only bazel makes now trouble, which can be called directly, or indirectly via pip3 install .
being in mediapipe  folder OR via pip3 wheel package generation with:
python3 setup.py bdist_wheel

I found also a similiar Issue which is closed = strange
https://github.com/google/mediapipe/issues/3211 
Can I simply get this file bazel wants ?
And than bypass the download process with the local file?

I would help me much :) 

Thanks folks.

I try my best to bypass the bazel "thing" but i didn't have succuess.
I simply want to have a mediapipe version + opencv which is running under latest Raspbian with ARM.

greetings.
```

### Complete Logs

```shell
WARNING: Download from http://mirror.tensorflow.org/github.com/bazelbuild/rules_closure/archive/cf1e44edb908e9616030cc83d085989b8e6cd6df.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/tensorflow/runtime/archive/2f6de37d68a4c69e2ff9eec3cebbf1369e496940.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
```

**Comment by schnitzlein** [comment]
continously getting those errors:

python3 setup.py bdist_wheel
/usr/lib/python3/dist-packages/setuptools/dist.py:475: UserWarning: The version specified ('dev') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  warnings.warn(
running bdist_wheel
running build
running build_py
running gen_protos
running generate_metadata_schema
DEBUG: Rule 'rules_foreign_cc' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "c2cdcf55ffaf49366725639e45dedd449b8c3fe22b54e31625eb80ce3a240f1e"
DEBUG: Repository rules_foreign_cc instantiated at:
  /usr/local/src/mediapipe/WORKSPACE:47:13: in <toplevel>
Repository rule http_archive defined at:
  /root/.cache/bazel/_bazel_root/dcc55e3eb2ef4c80ac4b81dec4ac8dda/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
ERROR: /root/.cache/bazel/_bazel_root/dcc55e3eb2ef4c80ac4b81dec4ac8dda/external/build_bazel_apple_support/crosstool/setup.bzl:69:32: name 'module_extension' is not defined
ERROR: error loading package '': in /root/.cache/bazel/_bazel_root/dcc55e3eb2ef4c80ac4b81dec4ac8dda/external/build_bazel_rules_swift/swift/extras.bzl: in /root/.cache/bazel/_bazel_root/dcc55e3eb2ef4c80ac4b81dec4ac8dda/external/build_bazel_apple_support/lib/repositories.bzl: Extension 'crosstool/setup.bzl' has errors
INFO: Elapsed time: 0,288s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)



But actually I have no idea how to setup bazel ?
Basically I want to build mediapipe on a linux system, which have Arm Cortex  CPU (Cortex-A72)

Unfortunatly the RPi4 with latest recommended Image comes with a Architecture "aarch64" instead of simply arm.

**Comment by schnitzlein** [comment]
Hi thanks for adding classification! :)

Any matching version constraints for Raspbian 11 (Bullseye) would be helpful.
Like:

- opencv 3.4
- mediapipe 10
- bazel ...
- python ...
- gcc/g++
- java
- protoc / protobuf
- other deps version numbers

Any constraint which guide me to a compile able mediapipe python wheel would be really helpful! 

Thanks in advance

**Comment by schnitzlein** [comment]
lscpu: shows me: aarch64 and ARM Cortex A72 CPU
maybe this helps?

how to convince bazel that this is POSIX ARM and it will find/download "things" and compile what I need ?

**Comment by kuaashish** [comment]
@schnitzlein,

Thank you for your interest in MediaPipe Solutions. For now, We do not have wheels for AArch64 in Python. However, Please be assured support of Raspberry Pi is already in our roadmap And We will let you know once an update becomes available.

**Comment by AiharaMahiru** [comment]
Has anyone tested that mediapipe is installed normally on rk3588?  The system architecture is Arm64.
